### PR TITLE
test: try reconnect Executor

### PR DESCRIPTION
**Pull request by JoanFM**
Migrated to [here](https://github.com/jina-ai/jina/pull/4941)

### CODE REVIEW

File: jina/serve/networking.py [hunk 1]
```diff
@@ -670,42 +674,55 @@ async def close(self):
...
+            error.code() == grpc.StatusCode.UNAVAILABLE
+            or error.code() == grpc.StatusCode.DEADLINE_EXCEEDED
+        ) and retry_i >= total_num_tries - 1:  # retries exhausted. if we land here it already failed once, therefore -1
             self._logger.debug(f'GRPC call failed, retries exhausted')
             from jina.excepts import InternalNetworkError
 
+            # after connection failure the gRPC `channel` gets stuck in a failure state for a few seconds
+            # removing and re-adding the connection (stub) is faster & more reliable than just waiting
+            if connection_list:
+                await asyncio.create_task(
```
Comment (JoanFM) [inline jina/serve/networking.py:712]: u should be able to simply do `await connection_list.remove_connection(current_address)`

File: jina/serve/networking.py [hunk 2]
```diff
@@ -754,6 +773,7 @@ def _send_discover_endpoint(
         self,
         connection: ConnectionStubs,
         timeout: Optional[float] = None,
+        connection_list: ReplicaList = None,  # needed for clean network error handling, but not crucial
```
Comment (JoanFM) [inline jina/serve/networking.py:776]:
```suggestion
        connection_list: Optional[ReplicaList] = None,  # needed for clean network error handling, but not crucial
```

File: tests/integration/runtimes/test_network_failures.py [hunk 3]
```diff
@@ -62,12 +63,8 @@ def _create_head(port, connection_list_dict, polling, retries=-1):
 def _send_request(gateway_port, protocol):
     """send request to gateway and see what happens"""
     c = Client(host='localhost', port=gateway_port, protocol=protocol)
-    return c.post(
```
Comment (JoanFM) [inline tests/integration/runtimes/test_network_failures.py:65]: to be equivalent, pass `return_responses = True` at Client initialization

File: jina/serve/networking.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 import contextlib
 import ipaddress
 import os
+import time
```
Comment (JoanFM) [inline jina/serve/networking.py:5]:
```suggestion
```

File: tests/integration/runtimes/test_network_failures.py [hunk 5]
```diff
@@ -163,6 +160,89 @@ async def test_runtimes_headless_topology(
...
+        worker_process.terminate()  # kill worker
+        worker_process.join()
+        assert not worker_process.is_alive()
+
+        # send request while Executor is not UP, WILL FAIL
+        p = multiprocessing.Process(target=_send_request, args=(gateway_port, protocol))
+        p.start()
+        p.join()
+        assert p.exitcode != 0
+        time.sleep(1)
```
Comment (JoanFM) [inline tests/integration/runtimes/test_network_failures.py:213]: remove time sleep

**Comment by JohannesMessner** [comment]
PR migrated to [here](https://github.com/jina-ai/jina/pull/4941)

**Pull request closed without merging**